std/experimental/allocator/mallocator.d

```diff
@@ -210,7 +225,8 @@ struct AlignedMallocator
     $(WEB msdn.microsoft.com/en-us/library/8z34s9c6(v=vs.80).aspx,
     $(D __aligned_malloc)) on Windows.
     */
-    version(Posix) @trusted @nogc
+    version(Posix)
+    @trusted @nogc nothrow
```

Comment:
Unfortunately there's no satisfactory way to make this both memory safe and `@nogc`.

For `ENOMEM`, return null per the allocator spec, problem handled.

`EINVAL` only happens for poor values of `a`. Per the assert, we can see that this function expects a good value of `a` as a precondition. Preconditions cannot be relied on for the memory safety guarantee. Throwing an exception on `EINVAL` provides memory safety put precludes `@nogc`, even in `onAssertError` is used, as it's not `@nogc` either. Returning `null` is not an option as it signals OOM.

I suggest leaving it as `@system` or using `code == EINVAL || assert(0);` as a stopgap measure until we've figured out `@nogc` exceptions.